packages/block-library/src/navigation/edit/index.js

```diff
@@ -109,12 +110,19 @@ function Navigation( {
 		layout: { justifyContent, orientation = 'horizontal' } = {},
 	} = attributes;
 
-	const [ areaMenu, setAreaMenu ] = useEntityProp(
-		'root',
-		'navigationArea',
-		'navigation',
-		navigationArea
-	);
+	let areaMenu,
+		setAreaMenu = noop;
+	// Navigation areas are deprecated and on their way out. Let's not perform
+	// the request unless we're in an environment where the endpoint exists.
+	if ( process.env.GUTENBERG_PHASE === 2 ) {
+		// eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
I'm not sure it's fine to ignore the rule of hooks. Won't this break other hooks since they need to be accessed in a known order?

Reply:
In this case it's okay because we can guarantee that `GUTENBERG_PHASE` won't change during the lifecycle of the application (it's a constant) and therefore the order in which the hooks are called won't change.

Reply:
Yeah, true, I guess that's ok then.